src/main/src/main/java/org/geoserver/security/KeyStoreProviderImpl.java

```diff
@@ -237,15 +234,14 @@ protected void assertActivatedKeyStore() throws IOException {
         char[] passwd = securityManager.getMasterPassword();
         try {
             ks = KeyStore.getInstance(KEYSTORETYPE);
-            if (getFile().exists()==false) { // create an empy one
+            if (getResource().getType() == Type.UNDEFINED) { // create an empy one
                 ks.load(null, passwd);
                 addInitialKeys();
-                FileOutputStream fos = new FileOutputStream(getFile());
+                OutputStream fos = getResource().out();
```

Comment:
Needs a try-with-resources block to ensure no resource leak occurs if an exception is thrown.